src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/ContainsKmerReadFilter.java

```diff
@@ -1,33 +1,52 @@
 package org.broadinstitute.hellbender.tools.spark.pathseq;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.broadinstitute.hellbender.engine.filters.ReadFilter;
-import org.broadinstitute.hellbender.tools.spark.sv.SVKmer;
 import org.broadinstitute.hellbender.tools.spark.sv.SVKmerShort;
 import org.broadinstitute.hellbender.tools.spark.sv.SVKmerizer;
 import org.broadinstitute.hellbender.utils.read.GATKRead;
 
-import java.util.Set;
-
 /**
- * Keep reads that DO NOT contain at least one kmer from a Set of SVKmerShorts
+ * Keep reads that do NOT contain one or more kmers from a set of SVKmerShorts
  */
 public class ContainsKmerReadFilter extends ReadFilter {
 
+    protected final Logger logger = LogManager.getLogger(this.getClass());
+
     private static final long serialVersionUID = 1L;
-    private Set<SVKmer> kmerLib;
-    private int kSize;
+    private static PSKmerCollection kmerLib = null;
+    private final int kSize, kmerCountThreshold;
 
-    public ContainsKmerReadFilter(final Set<SVKmer> kmer_lib, int kmer_size) {
-        kmerLib = kmer_lib;
-        kSize = kmer_size;
+    public ContainsKmerReadFilter(final String kmerLibPath, final int kmerCountThreshold) {
+        this.kmerCountThreshold = kmerCountThreshold;
+        if (kmerLib == null) {
```

Comment:
Done. I was actually getting infrequent null pointer exceptions on giant BAMs in Spark mode. I'll bet this was the culprit.